lib/Raven/ErrorHandler.php

```diff
@@ -11,203 +9,331 @@
+     *
+     * @internal
+     */
+    public function handleFatalError(array $error = null)
     {
-        unset($this->reservedMemory);
-
-        if (null === $error = error_get_last()) {
+        // If there is not enough memory that can be used to handle the error
+        // do nothing
+        if (null === self::$reservedMemory) {
             return;
         }
 
-        if ($this->shouldCaptureFatalError($error['type'])) {
-            $e = new \ErrorException(
-                @$error['message'],
-                0,
-                @$error['type'],
-                @$error['file'],
-                @$error['line']
-            );
-            $this->handleException($e, true);
+        self::$reservedMemory = null;
+        $exception = null;
+
+        if (null === $error) {
+            $error = error_get_last();
         }
-    }
 
-    public function shouldCaptureFatalError($type)
-    {
-        // Do not capture E_ERROR since those can be caught by userland since PHP 7.0
-        // E_ERROR should already be handled by the exception handler
-        // This prevents duplicated exceptions in PHP 7.0+
-        if (PHP_VERSION_ID >= 70000 && E_ERROR === $type) {
-            return false;
+        if (!empty($error) && $error['type'] & (E_ERROR | E_PARSE | E_CORE_ERROR | E_CORE_WARNING | E_COMPILE_ERROR | E_COMPILE_WARNING)) {
+            $exception = new \ErrorException(self::$errorLevelsDescriptions[$error['type']] . ': ' . $error['message'], 0, $error['type'], $error['file'], $error['line']);
         }
 
-        return $type & $this->fatal_error_types;
+        try {
+            if (null !== $exception) {
+                $this->handleException($exception);
+            }
+        } catch (\ErrorException $exception) {
+            // Ignore this re-throw
+        }
     }
 
     /**
-     * Register a handler which will intercept unhandled exceptions and report them to the
-     * associated Sentry client.
+     * Handles the given exception by capturing it through the Raven client and
+     * then forwarding it to another handler.
      *
-     * @param bool $call_existing Call any existing exception handlers after processing this instance
+     * @param \Exception|\Throwable $exception The exception to handle
      *
-     * @return \Raven\ErrorHandler
+     * @throws \Exception|\Throwable
+     *
+     * @internal
      */
-    public function registerExceptionHandler($call_existing = true)
+    public function handleException($exception)
     {
-        $this->old_exception_handler = set_exception_handler([$this, 'handleException']);
-        $this->call_existing_exception_handler = $call_existing;
+        $this->client->captureException($exception);
+
+        // If there is no exception handler that can run after the current one
+        // give back the exception to the native PHP handler
+        if (null === $this->previousExceptionHandler) {
+            throw $exception;
```

Comment:
Doesn't re-throwing change the stack trace of the exception?

Reply:
After a few tests it seems it doesn't